File: mealie/schema/household/equipment.py
from pydantic import UUID4, Field

from mealie.schema._mealie import MealieModel


class EquipmentRecipeSummary(MealieModel):
    slug: str
    name: str


class EquipmentOut(MealieModel):
    id: UUID4
    group_id: UUID4
    name: str
    slug: str
    category: str | None = None
    description: str | None = None
    image_source_url: str | None = None
    ai_enriched: bool = False
    has_image: bool = False
    image_version: str | None = None
    recipe_count: int = 0
    recipes: list[EquipmentRecipeSummary] = Field(default_factory=list)


class EquipmentCreate(MealieModel):
    name: str = Field(..., min_length=1, max_length=255)
    category: str | None = Field(None, max_length=120)
    description: str | None = Field(None, max_length=4000)


class EquipmentAICreateRequest(MealieModel):
    prompt: str = Field(..., min_length=2, max_length=8000)
    name: str | None = Field(None, max_length=255)


class EquipmentUpdate(MealieModel):
    category: str | None = Field(None, max_length=120)
    description: str | None = Field(None, max_length=4000)


class EquipmentImageURLRequest(MealieModel):
    url: str = Field(..., min_length=8, max_length=4000)

File: frontend/app/lib/api/types/equipment.ts
export interface EquipmentRecipeSummary {
  slug: string;
  name: string;
}

export interface Equipment {
  id: string;
  groupId: string;
  name: string;
  slug: string;
  category?: string | null;
  description?: string | null;
  imageSourceUrl?: string | null;
  aiEnriched: boolean;
  hasImage: boolean;
  imageVersion?: string | null;
  recipeCount: number;
  recipes: EquipmentRecipeSummary[];
}

export interface EquipmentUpdate {
  category?: string | null;
  description?: string | null;
}

export interface EquipmentCreate extends EquipmentUpdate {
  name: string;
}

export interface EquipmentAICreateRequest {
  prompt: string;
  name?: string | null;
}
